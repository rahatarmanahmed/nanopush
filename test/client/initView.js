const test = require('ava')

const initView = require('../../client/initView')
const { TOKEN } = require('../helpers/testData')

test('initView with notification permission denied', t => {
  const html = initView({
    notificationPermission: 'denied'
  })

  t.snapshot(html.toString())
})

test('initView without service worker support', t => {
  const html = initView({
    serviceWorkerSupported: false
  })

  t.snapshot(html.toString())
})

test('initView with notification permission not yet granted', t => {
  const html = initView({
    serviceWorkerSupported: true,
    notificationPermission: 'default'
  })

  t.snapshot(html.toString())
})

test('initView with token not yet ready', t => {
  const html = initView({
    serviceWorkerSupported: true,
    notificationPermission: 'granted'
  })

  t.snapshot(html.toString())
})

test('initView with unknown state', t => {
  const html = initView({
    serviceWorkerSupported: true,
    notificationPermission: 'default',
    token: TOKEN
  })

  t.snapshot(html.toString())
})
