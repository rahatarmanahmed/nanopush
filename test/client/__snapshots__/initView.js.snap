exports[`initView with notification permission denied 1`] = `"\"<p class=\\\"lh-copy o-60 measure\\\">Seems like you denied the request to show notifications. You\'re gonna have to reset that if you want to use this app.</p>\""`;

exports[`initView with notification permission not yet granted 1`] = `"\"<p class=\\\"lh-copy o-60 measure\\\">We\'ll get started once you accept the request to show notifications.</p>\""`;

exports[`initView with token not yet ready 1`] = `"\"<p class=\\\"lh-copy o-60 measure\\\">Registering web push notifications...</p>\""`;

exports[`initView with unknown state 1`] = `"\"<p class=\\\"lh-copy o-60 measure\\\">We\'ll get started once you accept the request to show notifications.</p>\""`;

exports[`initView without service worker support 1`] = `"\"<p class=\\\"lh-copy o-60 measure\\\">Your browser doesn\'t support web push notifications. Write to your local representative, or use something like Chrome or Firefox.</p>\""`;
